File: src/2562.find-the-array-concatenation-value.2698/solution.ts
function findTheArrayConcVal (nums: number[]): number {
    let result = 0;
    let left = 0;
    let right = nums.length - 1;
    while (left < right) {
        const str = `${nums[left++]}${nums[right--]}`;
        result += (+str);
    }
    if (left === right) {
        result += nums[left];
    }
    return result;
}

File: src/2562.find-the-array-concatenation-value.2698/solution.java
class Solution {
    public long findTheArrayConcVal(int[] nums) {
        long result = 0;
        StringBuilder sb = new StringBuilder();
        int left = 0;
        int right = nums.length - 1;
        while (left < right) {
            sb.append(nums[left]);
            sb.append(nums[right]);

            result += Long.parseLong(sb.toString());
            sb.delete(0, sb.length());

            left++;
            right--;
        }

        if (left == right) {
            result += nums[left];
        }

        return result;
    }
}